### PR TITLE
Quick Sort Algorithm with Rust

### PR DESCRIPTION
**Pull request by canberkkoc**
Solving #1616

### CODE REVIEW

File: code/sorting/quick_sort/quicksort.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+// This algorithm taken from https://github.com/servo/rust-quicksort
```
Comment (yatharthshahjpr) [inline code/sorting/quick_sort/quicksort.rs]: This code is copied so it cannot be merged !
U need to do certain corrections to it ! :)

File: code/sorting/quick_sort/quicksort.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+//! In-place sorting.
```
Comment (yatharthshahjpr) [inline code/sorting/quick_sort/quicksort.rs]: Kindly remove the above comments and add this comment  
/* Part of Cosmos by OpenGenus Foundation */
with every code you commit !

File: code/sorting/quick_sort/quicksort.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                arr.swap(p as usize, i as usize)
+            }
+            if compare(&*v, &arr[j as usize]) == Ordering::Equal {
+                q -= 1;
+                arr.swap(j as usize, q as usize)
+            }
+        }
+    }
+
+    arr.swap(i as usize, right as usize);
```
Comment (yatharthshahjpr) [inline code/sorting/quick_sort/quicksort.rs]: Improve the indentation of the code .....this is not upto the coding standards ! 👍

**Comment by AdiChat** [comment]
@yatharthshahjpr The code looks fine to me 👍

**Comment by AdiChat** [comment]
@yatharthshahjpr Kindly be patient 👍

File: code/sorting/quick_sort/quicksort.rs (hunk 2 above)
Comment (arnavb) [reply code/sorting/quick_sort/quicksort.rs]: The above comments do not need to be removed, but the header ones need to be added.

**Comment by yatharthshahjpr** [comment]
ok !

File: code/sorting/quick_sort/quicksort.rs (hunk 2 above)
Comment (canberkkoc) [reply code/sorting/quick_sort/quicksort.rs]: Shall i add Cosmos part at the begining of file ?

Comment (arnavb) [reply code/sorting/quick_sort/quicksort.rs]: It doesn't matter where it is, as long as it is near the top @canberkkoc.

File: code/sorting/quick_sort/quicksort.rs (hunk 3 above)
Comment (canberkkoc) [reply code/sorting/quick_sort/quicksort.rs]: This code is using at Servo i think their style is like that i don't get the problem at this 👍

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**